common/src/main/java/io/druid/math/expr/Evals.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.math.expr;
+
+import com.metamx.common.logger.Logger;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ */
+public class Evals
+{
+  private static final Logger log = new Logger(Evals.class);
+
+  public static boolean isConstant(Expr expr)
+  {
+    return expr instanceof DoubleExpr || expr instanceof LongExpr;
+  }
+
+  public static boolean isAllConstants(Expr... exprs)
+  {
+    return isAllConstants(Arrays.asList(exprs));
+  }
+
+  public static boolean isAllConstants(List<Expr> exprs)
+  {
+    for (Expr expr : exprs) {
+      if (!isConstant(expr)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  public static Expr toConstant(Number eval)
+  {
+    if (eval instanceof Float || eval instanceof Double) {
+      return new DoubleExpr(eval.doubleValue());
+    }
+    return new LongExpr(eval.longValue());
+  }
+
+  public static Expr binaryOp(BinaryOpExprBase binary, Expr left, Expr right)
+  {
+    try {
+      return binary.getClass()
+                   .getDeclaredConstructor(String.class, Expr.class, Expr.class)
```

Comment:
this means that every subclass of BinaryOpExprBase should have a constr matching this signature, do we need to mention this in javadoc for class BinaryOpExprBase ?